File: CallX2/app/src/main/java/com/callx/app/adapters/MemberSelectAdapter.java
package com.callx.app.adapters;
import android.view.*;
import android.widget.CheckBox;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.callx.app.R;
import com.callx.app.models.User;
import de.hdodenhof.circleimageview.CircleImageView;
import java.util.List;
import java.util.Set;
public class MemberSelectAdapter
        extends RecyclerView.Adapter<MemberSelectAdapter.VH> {
    private final List<User> users;
    private final Set<String> selected;
    public MemberSelectAdapter(List<User> users, Set<String> selected) {
        this.users = users; this.selected = selected;
    }
    @NonNull @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_member_select, parent, false);
        return new VH(v);
    }
    @Override public void onBindViewHolder(@NonNull VH h, int pos) {
        User u = users.get(pos);
        h.tvName.setText(u.name == null ? "User" : u.name);
        if (u.photoUrl != null && !u.photoUrl.isEmpty()) {
            Glide.with(h.itemView.getContext()).load(u.photoUrl).into(h.ivAvatar);
        } else h.ivAvatar.setImageResource(R.drawable.ic_person);
        h.cb.setOnCheckedChangeListener(null);
        h.cb.setChecked(selected.contains(u.uid));
        h.cb.setOnCheckedChangeListener((b, c) -> {
            if (c) selected.add(u.uid); else selected.remove(u.uid);
        });
        h.itemView.setOnClickListener(v -> h.cb.setChecked(!h.cb.isChecked()));
    }
    @Override public int getItemCount() { return users.size(); }
    static class VH extends RecyclerView.ViewHolder {
        TextView tvName;
        CircleImageView ivAvatar;
        CheckBox cb;
        VH(View v) {
            super(v);
            tvName = v.findViewById(R.id.tv_name);
            ivAvatar = v.findViewById(R.id.iv_avatar);
            cb = v.findViewById(R.id.cb_select);
        }
    }
}

File: CallX2/feature-chat/src/main/java/com/callx/app/adapters/MemberSelectAdapter.kt
package com.callx.app.adapters

import android.view.*
import android.widget.CheckBox
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.callx.app.chat.R
import com.callx.app.models.User
import de.hdodenhof.circleimageview.CircleImageView

class MemberSelectAdapter(
    private val users: List<User>,
    private val selected: MutableSet<String>
) : RecyclerView.Adapter<MemberSelectAdapter.VH>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_member_select, parent, false)
        return VH(v)
    }

    override fun onBindViewHolder(h: VH, pos: Int) {
        val u = users[pos]
        h.tvName.text = u.name ?: "User"
        val avatarUrl = u.thumbUrl?.takeIf { it.isNotEmpty() } ?: u.photoUrl
        if (!avatarUrl.isNullOrEmpty()) Glide.with(h.itemView.context).load(avatarUrl).circleCrop().into(h.ivAvatar)
        else h.ivAvatar.setImageResource(R.drawable.ic_person)

        h.cb.setOnCheckedChangeListener(null)
        h.cb.isChecked = selected.contains(u.uid)
        h.cb.setOnCheckedChangeListener { _, checked ->
            if (checked) u.uid?.let { selected.add(it) } else u.uid?.let { selected.remove(it) }
        }
        h.itemView.setOnClickListener { h.cb.isChecked = !h.cb.isChecked }
    }

    override fun getItemCount() = users.size

    class VH(v: View) : RecyclerView.ViewHolder(v) {
        val tvName:   TextView        = v.findViewById(R.id.tv_name)
        val ivAvatar: CircleImageView = v.findViewById(R.id.iv_avatar)
        val cb:       CheckBox        = v.findViewById(R.id.cb_select)
    }
}
